core/src/main/java/com/yahoo/imapnio/async/client/ImapAsyncClient.java

```diff
@@ -210,16 +214,16 @@ public void operationComplete(final Future<? super Void> future) {
                     }
                     final int sessionId = sessionCount.incrementAndGet();
                     pipeline.addLast(ImapClientConnectHandler.HANDLER_NAME,
-                            new ImapClientConnectHandler(sessionFuture, LoggerFactory.getLogger(ImapAsyncSessionImpl.class), sessionId));
+                            new ImapClientConnectHandler(sessionFuture, LoggerFactory.getLogger(ImapAsyncSessionImpl.class), logOpt, sessionId));
 
-                    if (logger.isDebugEnabled()) {
+                    if (isSessionDebugOn || logger.isDebugEnabled()) {
                         logger.debug("{}, connect operationComplete with success, imapServerUri={}", sessionId, serverUri.toASCIIString());
```

Comment:
would like to see the actual request/response payload being sent over the wire, can you check how to do this

Reply:
Done

Reply:
https://docs.google.com/document/d/1kxv3TFaekDvdj9Zveh73x9_Z1RTpdNG1Aybp4sE06HA/edit?usp=sharing